spring-kafka/src/main/java/org/springframework/kafka/core/RoutingKafkaTemplate.java

```diff
@@ -81,8 +81,7 @@ public RoutingKafkaTemplate(Map<Pattern, ProducerFactory<Object, Object>> factor
 		if (producerFactory == null) {
 			for (Entry<Pattern, ProducerFactory<Object, Object>> entry : this.factoryMatchers.entrySet()) {
 				if (entry.getKey().matcher(topic).matches()) {
-					producerFactory = entry.getValue();
-					this.factoryMap.put(topic, producerFactory);
+					producerFactory = this.factoryMap.computeIfAbsent(topic, key -> entry.getValue());
```

Comment:
No, I mean the whole `for ()` logic and pattern matching should go to that lambda instead of all of these `if`, `put` and `break`. The lambda would have a return for matching value.